### ISSUE TITLE
How to make a custom layer with a custom backward function like updateGradInput in Torch?

### ISSUE THREAD

**Issue by lllyasviel** [fchollet/keras#6604]
I am familiar with torch but new to Keras. Recently I have to implement a custom layer in Keras but I can not find a way to implement the "backward function", such as "updateGradInput" in Torch. When I write a custom layer the only thing I can modify is "def call(self, x)" for forward predict.
Maybe this is a simple problem. But is has already confused me for more than two weeks. The deadline of our project  is approaching.
Thank you very much!
For example, if I write a custom layer in torch with a "updateGradInput", which always return a matrix filled with zero, then all layers below to the custom layer will not get any gradient. However, I can not find a way to write custom "updateGradInput" in Keras.

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by lllyasviel** [comment]
Here is what I want to do.
There is a layer with input_shape (None,5,5,512)
I want to:
forward:
1. reshape input A the shape into (None,25,512)
2. copy it to a new matrix B
3. transpose B into (None,512,25)
4.dot B and A to get a output of shape(None,512,512)
5.reshape it into (None,512,512,1)

**Comment by PantherYan** [comment]
Have you solved this ? @lllyasviel

**Comment by MingleiLI** [comment]
Have you solved this? @lllyasviel

**Comment by houkai** [comment]
Have you solved this? @lllyasviel
